src/pkcs15init/pkcs15-myeid.c

```diff
@@ -349,19 +352,33 @@ myeid_new_file(sc_profile_t *profile, sc_card_t *card,
 	sc_file_t *file;
 	sc_path_t *p;
 	char name[64];
-	const char *tag;
+	const char *tag = NULL;
 	int r;
 
 	LOG_FUNC_CALLED(card->ctx);
-	if (type == SC_PKCS15_TYPE_PRKEY_RSA || type == SC_PKCS15_TYPE_PRKEY_EC)
+	switch (type) {
+	case SC_PKCS15_TYPE_PRKEY_RSA:
+	case SC_PKCS15_TYPE_PRKEY_EC:
```

Comment:
The only problem left is what I described earlier:

> It seems that TokenInfo is not written after creating a secret key that adds the supportedAlgorithms.
> what would be the way to signal sc_pkcs15init_unbind to update TokenInfo regardless of profile->pkcs15.do_last_update value when the TokenInfo contents are "dirty"?

After the TokenInfo on card is updated properly, this pull request should be ready for merge unless additional issues are found.